osf_tests/test_elastic_search.py

```diff
@@ -62,6 +71,176 @@ def wrapped(*args, **kwargs):
         return wrapped
     return test_wrapper
 
+class TestCollectionsSearch(OsfTestCase):
```

Comment:
A lot of these tests are mostly the same, but do the opposite of each other. Consider collating them in such a way as to minimize set-up time without testing any less functionality. (e.g. combining `test_submissions_turned_private_are_deleted_from_index` and `test_submissions_turned_public_are_added_to_index` to go from public -> private -> public again, checking ES in between, etc.)